Server/Services/RatingService.cs

```diff
@@ -75,15 +76,78 @@ private int CalculateScoreDelta(double averageWinningRating, double averageLosin
             return result;
         }
 
-        private Dictionary<Player, GameResult> SplitTeams(IEnumerable<MatchPlayer> matchPlayers, ushort winnerTeam)
+        #endregion
+
+        #region Overthrow
+
+        private Dictionary<string, List<int>> _scores = new Dictionary<string, List<int>>()
         {
-            var result = new Dictionary<Player, GameResult>();
-            foreach (var matchPlayer in matchPlayers)
-                result.Add(matchPlayer.Player, matchPlayer.Team == winnerTeam ? GameResult.Winner : GameResult.Loser);
+            { "forest_solo", new List<int> { 30, 25, 18, 11, 4, -4, -11, -18, -25, -30 } },
+            { "core_quartet", new List<int> { 30, 18, 6, -6, -18, -30 } },
+            { "desert_duo", new List<int> { 30, 15, 0, -15, -30 } },
+            { "temple_quartet", new List<int> { 30, 10, -10, -30 } },
+            { "mines_trio", new List<int> { 30, 0, -30 } },
+            { "desert_quintet", new List<int> { 30, 0, -30 } },
+            { "desert_octet", new List<int> { 30, 0, -30 } }
+        };
+        public Dictionary<ulong, PlayerRatingChange> RecordRankedMatchOverwatch(IEnumerable<MatchPlayer> matchPlayers, string mapName)
+        {
+            var result = new Dictionary<ulong, PlayerRatingChange>();
+
+            var teams = matchPlayers
+                .GroupBy(t => t.Team)
+                .OrderByDescending(g => g.Sum(p => p.Kills))
+                .ThenBy(g => g.Max(p => p.LastKill))
+                .ToDictionary(g => g.Key, g => g.Select(mp => mp.Player).ToList());
+            var teamScores = teams.Keys
+                .Zip(_scores[mapName], (k, v) => new { k, v })
+                .ToDictionary(x => x.k, x => x.v);
+
+            foreach (var team in teams)
+            {
+                var averageCurrentTeamRating = team.Value
+                    .Average(x => x.PlayerOverthrowRating.FirstOrDefault(p => p.MapName == mapName)?.Rating);
+                var averageOtherTeamRating = teams
+                    .Where(x => x.Key != team.Key)
+                    .SelectMany(t => t.Value)
+                    .Average(x => x.PlayerOverthrowRating.FirstOrDefault(p => p.MapName == mapName)?.Rating);
+
+                var scoreDelta = CalculateScoreDelta(averageCurrentTeamRating ?? PlayerOverthrowRating.DefaultRating,
+                                          averageOtherTeamRating ?? PlayerOverthrowRating.DefaultRating);
+                var ratingChange = teamScores[team.Key] > 0  ? teamScores[team.Key] + scoreDelta : teamScores[team.Key] - scoreDelta;
+
+                GetPlayersChangeOverthrow(team.Value, ratingChange, result, mapName);
```

Comment:
I don't like passing a reference just to mutate it. And in general that method is pretty simple and depends on the context too much, so I don't see much point for it to be extracted to it's own method
```suggestion
                foreach (var player in team.Value)
                {
                    var ratingRecord = player.PlayerOverthrowRating.FirstOrDefault(x => x.MapName == mapName);
                    var oldRating = ratingRecord.Rating;
                    ratingRecord.Rating += scoreChange;
                    result[player.SteamId] = new PlayerRatingChange() { Old = oldRating, New = ratingRecord.Rating };
                }
```